src/retro_achievements.cpp

```diff
@@ -0,0 +1,1653 @@
+// TODO: use some tool to simulate slow internet connection and try this
+
+#include "sokol_gfx.h"
+
+extern "C" {
+#define CIMGUI_DEFINE_ENUMS_AND_STRUCTS
+#include "cimgui.h"
+const char* se_get_pref_path();
+void se_push_disabled();
+void se_pop_disabled();
+void se_text(const char* fmt, ...);
+void se_boxed_image_dual_label(const char* title, const char* description, const char* icon,
+                               sg_image image, int flags, ImVec2 uv0, ImVec2 uv1);
+bool se_button(const char* label, ImVec2 size);
+const char* se_localize_and_cache(const char* input_str);
+ImFont* se_get_mono_font();
+void se_emscripten_flush_fs();
+#include "retro_achievements.h"
+ra_config_t* se_get_ra_config();
+}
+
+#include "https.hpp"
+#include "IconsForkAwesome.h"
+#include "rc_client.h"
+#include "rc_consoles.h"
+#include "sb_types.h"
+#include "sokol_time.h"
+
+#include <atomic>
+#include <cassert>
+#include <chrono>
+#include <condition_variable>
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <memory>
+#include <mutex>
+#include <sstream>
+#include <thread>
+#include <unordered_map>
+#include <vector>
+
+#define STBI_ONLY_PNG
+#include "stb_image.h"
+#include "stb_image_write.h"
+
+const uint32_t notification_end_frame = 60 * 4;
+const uint32_t notification_fade_frame = 60 * 3;
+#if defined(SE_PLATFORM_IOS) || defined(SE_PLATFORM_ANDROID)
+const bool only_one_notification = true;
```

Comment:
What are these for?

Reply:
I figured that on mobile if you show too many notifications the screen is going to get filled with them, so I made it so only one is shown at a time

Reply:
Okay, makes sense. However, you should use...

gui_state.ui_type == SE_UI_ANDROID || gui_state.ui_type == SE_UI_IOS

...instead of the preprocessor macros. That way things will behave the same under CI and eventually if we get it working on the web (where running on iOS or Android is a runtime characteristic, rather than a build one).